flake.nix

```diff
@@ -49,26 +45,50 @@
         let
           pkgs = pkgsFor system;
 
-          rust = pkgs.rust-bin.fromRustupToolchainFile ./rust-toolchain;
+          rust = fenix.packages.${system}.fromToolchainFile {
+            file = ./rust-toolchain;
+            sha256 = "sha256-6PfBjfCI9DaNRyGigEmuUP2pcamWsWGc4g7SNEHqD2c=";
+          };
 
-          naersk-lib = naersk.lib."${system}".override {
+          buildRustPackage = (pkgs.makeRustPlatform {
             cargo = rust;
             rustc = rust;
-          };
+          }).buildRustPackage;
         in
         rec {
           # `nix build`
-          packages.${name} = naersk-lib.buildPackage {
+          packages.${name} = buildRustPackage rec {
+            inherit name;
             pname = name;
-            root = ./.;
+            src = nixpkgs.lib.cleanSource ./.;
```

Comment:
While you're at it: would you mind to properly filter out the Cargo sources either using `builtins.path` or (if required) [`nix-filter`](https://github.com/numtide/nix-filter)?

Reply:
To keep with `@roberth` addition is easier than substraction relating to hermeticiry. In that light my personal preference is putting things into subfolders in such cases or bear the cost of an occasional extra rebuild (already reasonably reduced by `nixpkgs.lib.cleanSource`).

Means: I think excessive filtering puts an (almost) invisible layer between the spectator and the obvious file system layout and is a bad for transparency.

Reply:
Looking at it again, let's also filter out `flake.nix` & `flake.lock` witch are pure  nix concernts, but almost never the concern of a rust app :smile: